### PR TITLE
Display last modification date in listing + provides a way to sort by modification date

### PR DESCRIPTION
**Pull request by boastful-squirrel**
This addresses #36 

If there's something I should improve, please tell me :)

### CODE REVIEW

**Comment by boastful-squirrel** [comment]
screenshot 2019-02-17 at 00 05 20

**Comment by boastful-squirrel** [comment]
screenshot 2019-02-17 at 09 18 31

I tried to mimick `ls -l`'s result when it comes to the format.

**Comment by boastful-squirrel** [comment]
I'll update the README tonight ;)

File: src/listing.rs [hunk 1]
```diff
@@ -260,3 +306,39 @@ pub fn directory_listing<S>(
...
+fn convert_to_utc(src_time: Option<SystemTime>) -> (String, String) {
+    match src_time {
+        Some(time) => {
+            let date_time = DateTime::<Utc>::from(time);
+            (
+                date_time.format("%e %b").to_string(),
+                date_time.format("%R").to_string(),
+            )
+        }
+        None => ("".to_string(), "".to_string()),
```
Comment (svenstaro) [inline src/listing.rs]: Can't this be done more simply using `unwrap_or_default()` or such?

File: src/listing.rs [hunk 2]
```diff
@@ -260,3 +306,39 @@ pub fn directory_listing<S>(
...
+fn humanize_duration(src_time: Option<SystemTime>) -> String {
+    match src_time {
+        Some(std_time) => match SystemTime::now().duration_since(std_time) {
+            Ok(from_now) => match Duration::from_std(from_now) {
+                Ok(duration) => HumanTime::from(duration).to_text_en(Accuracy::Rough, Tense::Past),
+                Err(_) => "".to_string(),
+            },
+            Err(_) => "".to_string(),
+        },
+        None => "".to_string(),
```
Comment (svenstaro) [inline src/listing.rs]: Same as above, can't we do all of these default values using `unwrap_or_default()`?

Comment (boastful-squirrel) [reply src/listing.rs]: I'll try :)

File: src/listing.rs (hunk 1 above)
Comment (boastful-squirrel) [reply src/listing.rs]: I'll try :)

File: src/listing.rs [hunk 3]
```diff
@@ -260,3 +306,39 @@ pub fn directory_listing<S>(
...
+        None => ("".to_string(), "".to_string()),
+    }
+}
+
+/// Converts a SystemTime to a string readable by a human,
+/// i.e. calculates the duration between now() and the given SystemTime,
+/// and gives a rough approximation of the elapsed time since
+///
+/// If no SystemTime was given, returns an empty string
+fn humanize_duration(src_time: Option<SystemTime>) -> String {
```
Comment (svenstaro) [inline src/listing.rs]: Wrong name, I think, as this is not a duration but a point in time. So humanize_systemtime?

On a side note, wouldn't it be cool actually put a function like this into chrono_humanize itself?

Comment (boastful-squirrel) [reply src/listing.rs]: True, that would be a good addition to the crate

**Comment by svenstaro** [comment]
Excellent work. The conversion still looks a little bit weird but I think it's better than before. I'll take it!

**Pull request merged**